pkg/filesystem/path/windows_parser.go

```diff
@@ -0,0 +1,152 @@
+package path
+
+import (
+	"strings"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type windowsParser struct {
+	path string
+}
+
+type windowsRelativeParser struct {
+	path string
+}
+
+// NewWindowsParser creates a Parser for Windows paths that can be used in Resolve.
+func NewWindowsParser(path string) (Parser, error) {
+	// Windows paths are generally passed to system calls that
+	// accept C strings. There is no way these can accept null
+	// bytes.
+	if strings.ContainsRune(path, '\x00') {
+		return nil, status.Error(codes.InvalidArgument, "Path contains a null byte")
+	}
+	return &windowsParser{path}, nil
+}
+
+// MustNewWindowsParser is identical to NewWindowsParser, except that it panics
+// upon failure.
+func MustNewWindowsParser(path string) Parser {
+	parser, err := NewWindowsParser(path)
+	if err != nil {
+		panic(err)
+	}
+	return parser
+}
+
+func (p windowsParser) ParseScope(scopeWalker ScopeWalker) (next ComponentWalker, remainder RelativeParser, err error) {
+	path := strings.Replace(p.path, "/", "\\", -1)
```

Comment:
Any way we can implement this without copying the incoming string?

Reply:
Sure. I can try that, more and more code need to have work on "separator" rather than "slash" so I can give that a stab here too.

Reply:
We practically had everything in place already, this was a small change following the other refactorings for path separators :+1:.